clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -218,8 +216,7 @@ private void poll(long timeout, long now, boolean executeDelayedTasks) {
         checkDisconnects(now);
 
         // execute scheduled tasks
-        if (executeDelayedTasks)
-            delayedTasks.poll(now);
+        delayedTasks.poll(now);
```

Comment:
@becketqin Apologies. Looks like I wiped out the patch just as you were reviewing it. If you have time, would you mind looking at the update?